### DISCUSSION TITLE
Nullable tuple deconstruction

### DISCUSSION THREAD

**Discussion by Thaina** [dotnet/csharplang#2680] [Language Ideas]
Should this be possible?

```C#
(long x,(int y,byte z,string w,object o)?) complexTuple;

var (x,(y,z,w,o)) = complexTuple; // can deconstruct to (long,(int?,byte?,string,object))
```

**Comment by alrz** [comment]
you could use `if (complexTuple is var (x,(y,z,w,o)))` instead?

**Comment by Thaina** [comment]
@alrz What will be the type of `y` ?

Will it be `int` or `int?` ?

**Comment by yaakov-h** [comment]
It will be `int`. Furthermore, that deconstruction will only occur if the inner tuple is non-null / `.HasValue == true`.
